pygmt/src/shift_origin.py

```diff
@@ -1,33 +1,57 @@
 """
 shift_origin - Shift plot origin in x and/or y directions.
 """
+from __future__ import annotations
 
 from pygmt.clib import Session
 
 
-def shift_origin(self, xshift=None, yshift=None):
-    """
+def shift_origin(
+    self, xshift: float | str | None = None, yshift: float | str | None = None
+) -> None:
+    r"""
     Shift plot origin in x and/or y directions.
 
-    This method shifts the plot origin relative to the current origin
-    by (*xshift*, *yshift*). Optionally, append the length unit (**c**,
-    **i**, or **p**). Default unit if not given is **c** for centimeters.
+    This method shifts the plot origin relative to the current origin by *xshift* and
+    *yshift* in x and y directions, respectively. Optionally, append the length unit
+    (**c** for centimeters, **i** for inches, or **p** for points) to the shifts.
+    Default unit if not explicitly given is **c**, but can be changed to other units
+    via :gmt-term:`PROJ_LENGTH_UNIT`.
+
+    For *xshift*, a special character **w** can also be used, which represents the
+    bounding box **width** of the previous plot. The full syntax is
+    [[±][*f*]\ **w**\ [/\ *d*\ ]±]\ *xoff*, where optional signs, factor *f* and
+    divisor *d* can be used to compute an offset that may be adjusted further by
+    ±\ *xoff*. For example, assuming that the previous plot has a width of 10
+    centimeters, here are some example values for *xshift*:
```

Comment:
Maybe we can shorten this?
```suggestion
    ±\ *xoff*. Assuming that the previous plot has a width of 10
    centimeters, here are some example values for *xshift*:
```